packages/legend-extension-dsl-data-space/src/__lib__/query/DSL_DataSpace_LegendQueryNavigation.ts

```diff
@@ -35,20 +35,33 @@ export enum DATA_SPACE_QUERY_CREATOR_QUERY_PARAM_TOKEN {
   CLASS_PATH = 'class',
 }
 
+export enum DATA_SPACE_TEMPLATE_QUERY_VIEWER_ROUTE_PATTERN_TOKEN {
+  GAV = 'gav',
+  DATA_SPACE_PATH = 'dataSpacePath',
+  TEMPLATE_QUERY_TITLE = 'templateQueryTitle',
+}
+
 export type DataSpaceQueryCreatorPathParams = {
   [DATA_SPACE_QUERY_CREATOR_ROUTE_PATTERN_TOKEN.GAV]: string;
   [DATA_SPACE_QUERY_CREATOR_ROUTE_PATTERN_TOKEN.DATA_SPACE_PATH]: string;
   [DATA_SPACE_QUERY_CREATOR_ROUTE_PATTERN_TOKEN.EXECUTION_CONTEXT]: string;
   [DATA_SPACE_QUERY_CREATOR_ROUTE_PATTERN_TOKEN.RUNTIME_PATH]?: string;
 };
 
+export type DataSpaceTemplateQueryViewerPathParams = {
+  [DATA_SPACE_TEMPLATE_QUERY_VIEWER_ROUTE_PATTERN_TOKEN.GAV]: string;
+  [DATA_SPACE_TEMPLATE_QUERY_VIEWER_ROUTE_PATTERN_TOKEN.DATA_SPACE_PATH]: string;
+  [DATA_SPACE_TEMPLATE_QUERY_VIEWER_ROUTE_PATTERN_TOKEN.TEMPLATE_QUERY_TITLE]: string;
+};
+
 export type DataSpaceQueryEditorQueryParams = {
   [DATA_SPACE_QUERY_CREATOR_QUERY_PARAM_TOKEN.CLASS_PATH]?: string | undefined;
 };
 
 export const DATA_SPACE_QUERY_ROUTE_PATTERN = Object.freeze({
   SETUP: `/dataspace`,
   CREATE: `/dataspace/:${DATA_SPACE_QUERY_CREATOR_ROUTE_PATTERN_TOKEN.GAV}/:${DATA_SPACE_QUERY_CREATOR_ROUTE_PATTERN_TOKEN.DATA_SPACE_PATH}/:${DATA_SPACE_QUERY_CREATOR_ROUTE_PATTERN_TOKEN.EXECUTION_CONTEXT}/:${DATA_SPACE_QUERY_CREATOR_ROUTE_PATTERN_TOKEN.RUNTIME_PATH}?`,
+  TEMPLATE_QUERY: `/template/dataspace/:${DATA_SPACE_TEMPLATE_QUERY_VIEWER_ROUTE_PATTERN_TOKEN.GAV}/:${DATA_SPACE_TEMPLATE_QUERY_VIEWER_ROUTE_PATTERN_TOKEN.DATA_SPACE_PATH}/:${DATA_SPACE_TEMPLATE_QUERY_VIEWER_ROUTE_PATTERN_TOKEN.TEMPLATE_QUERY_TITLE}`,
```

Comment:
not sure this URL makes sense. wouldnt
`dataspace/gavDataspaceCreatore/datspacePath/template/{templateTitle}`
make more sense.